### PR TITLE
Caff 9 - Add Navsat Transform Node

### PR DESCRIPTION
**Pull request by erickmu1**
The `navsat_transform_node` is called in `odom.launch` by `caffeine_gazebo.launch`. The GPS odometry is published on `caffeine/odometry/gps` and works as expected with keyboard teleop (ie. no obvious drift that seems like an error and has correct orientation).
- You will notice that when you spin Caffeine in circles (around itself), the GPS odom does not spin, it stays in place, this is because there is no orientation data present in GPS coords. Moreover, the IMU sent to the `navsat_transform` node is only used for initialization (ie. replaces the manual use of the `datum` service) and hence cannot be used for orientation
- Orientation will be addressed when fusing GPS odom with local odom to get a "map" odom

Looking at the TF tree (`rosrun rqt_tf_tree rqt_tf_tree`) we see the UTM frame being published as a child of the odom frame. Once we start fusing GPS and local odometry, we will publish the map frame and the UTM frame will be a child of it.

The reason for why I set all `<driftVelocity>` to "0" is because the GPS odom seemed to clearly drift in one direction when Caffeine was static, despite the GPS values staying relatively in place (except for altitude, but that was avoided using the `zero_altitude` param). So for now I decided to remove them, giving us a more ideal GPS. In the future, when testing more realistic conditions, this attribute can be re-adjusted so that we observe how robust our set up is (it is a bit early now to try to add too many un-idealities).

Lastly, a thought that came to mind while working on this is "why not add the `lib_hector_gazebo` magnetometer plugin?" I finally decided against that since the current configuration of the imu plugin respects the ENU convention with a fixed yaw offset, and thus is compatible with `robot_localization`. Moreover, the magnetometer plugin publishes a `Vector3Stamped`, which is outdated and thus an extra node would be needed to publish it as a `MagField` message (I don't want to do this since as argued, is unnecessary for use with `robot_localization`).

**Note:** Publishing the map frame will occur during #10 , so at this moment the functionality added may feel a bit incomplete.

### CODE REVIEW

File: odom/odom/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+#############
+
+## Add gtest based cpp test target and link libraries
+# catkin_add_gtest(${PROJECT_NAME}-test test/test_odom.cpp)
+# if(TARGET ${PROJECT_NAME}-test)
+#   target_link_libraries(${PROJECT_NAME}-test ${PROJECT_NAME})
+# endif()
+
+## Add folders to be run by python nosetests
+# catkin_add_nosetests(test)
```
Comment (kajananchinniah) [inline odom/odom/CMakeLists.txt:204]: We should cleanup this file to include only things that are necessary

File: odom/odom/package.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0"?>
```
Comment (kajananchinniah) [inline odom/odom/package.xml:1]: We should clean up this file as well

File: odom/odom/launch/odom.launch [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+            <remap from="imu/data" to="imu/data"/>
+            <remap from="odometry/filtered" to="odom"/>
+            <remap from="gps/fix" to="gps/fix"/>
+
+            <rosparam command="load" file="$(find odom)/config/caffeine_navsat.yaml"/>
+        </node>
+
+    </group>
+
+</launch>
```
Comment (kajananchinniah) [inline odom/odom/launch/odom.launch:18]: One thing we should note is CAFF-8 also uses `odom.launch`. Should they both remain in the file or should we create two separate files (e.g. `local_odom.launch` and `navstat_odom.launch`)

Comment (erickmu1) [reply odom/odom/launch/odom.launch:18]: They should all remain in the same file, `odom.launch` should run all ekf and navsat nodes we need

**Pull request merged**